pkg/cloud/gcp/actuators/machine/reconciler.go

```diff
@@ -663,15 +663,38 @@ func (r *Reconciler) ensureInstanceGroup(instanceGroupName string) error {
 	return nil
 }
 
+// ensureCorrectNetworkAndSubnetName makes sure that we are working with the correct network and subnet.
+// In a case of VPC, we have to look whether the expect name of the network and subnet resource
+// matches the one, that is actually up in the cluster.
+func (r *Reconciler) ensureCorrectNetworkAndSubnetName() (string, string, error) {
```

Comment:
So this function, checks if any of the networks match the "expected" format based on our installer expectations, else we just return the network of the first interface? Seems reasonable, but why not always go for the first network interface? Any thoughts on when this might be different?

Reply:
Depends what we would prefer. The thing you are suggesting makes sense as well.
I just wanted to split these 2 options: 1. VPC 2. Not VPC. I think the way it is now is also more explicit about what we expect the network and subnet to be on a regular cluster.